pkg/storage/ingest/fetcher.go

```diff
@@ -466,14 +465,14 @@ func (g offsetRange) numOffsets() int64 {
 	return g.end - g.start
 }
 
-func findGapsInRecords(records []*kgo.Record, lastReturnedOffset int64) []offsetRange {
+func findGapsInRecords(records kgo.Fetches, lastReturnedOffset int64) []offsetRange {
```

Comment:
[nit] Just to keep consistency.

```suggestion
func findGapsInRecords(records kgo.Fetches, lastSeenOffset int64) []offsetRange {
```